### PR TITLE
Machine list changes for IIS & SQL task

### PR DESCRIPTION
**Pull request by chshrikh**

### CODE REVIEW

File: Tasks/IISWebAppDeployment/DeployIISWebApp.ps1 [hunk 1]
```diff
@@ -32,6 +36,8 @@
 
 Write-Verbose "Entering script DeployIISWebApp.ps1" -Verbose
 
+ # Constants #
+$defaultTestCertificateValue = "true"
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/DeployIISWebApp.ps1]: I believe this is not required

File: Tasks/IISWebAppDeployment/DeployIISWebApp.ps1 [hunk 2]
```diff
@@ -47,7 +53,16 @@ else
     $hostName = $hostNameWithOutSNI
 }
 
+# If testCertificate option is not set, set it to default 
+if([string]::IsNullOrEmpty($testCertificate))
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/DeployIISWebApp.ps1]: I believe this is not required

File: Tasks/IISWebAppDeployment/task.json [hunk 3]
```diff
@@ -1,371 +1,409 @@
...
+      "type": "string",
+      "label": "User Name",
+      "required": true,
+      "groupName": "website",
+      "defaultValue": "",
+      "visibleRule": "WebSitePhysicalPathAuth = WebSiteWindowsAuth",
+      "helpMarkDown": "User name for accessing the website's physical path."
+    },
+    {
+      "name": "WebSiteAuthUserPassword",
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/task.json]: This is tough to understand what changes happened. Can you fix indentation and update PR

File: Tasks/IISWebAppDeployment/task.json [hunk 4]
```diff
@@ -1,371 +1,409 @@
...
+  ],
+  "author": "Microsoft Corporation",
+  "version": {
+    "Major": 1,
+    "Minor": 0,
+    "Patch": 3
+  },
+  "demands": [
+  ],
+  "minimumAgentVersion": "1.88.0",
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/task.json]: Change the min agent version to 90

File: Tasks/IISWebAppDeployment/DeployIISWebApp.ps1 [hunk 5]
```diff
@@ -1,5 +1,9 @@
 ﻿param (
     [string]$environmentName,
+	[string]$adminUserName,
+    [string]$adminPassword,
+    [string]$winrmProtocol,
+    [string]$testCertificate,
```
Comment (mvvsubbu) [inline Tasks/IISWebAppDeployment/DeployIISWebApp.ps1]: In Jason TestCert but here it is TestCertificate. I think they should be same except for casing.

File: Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1 [hunk 6]
```diff
@@ -14,8 +18,14 @@
     [string]$deployInParallel    
     )
 
+	 # Constants #
+$defaultTestCertificateValue = "true"
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1]: Remove this.. it is not required

File: Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1 [hunk 7]
```diff
@@ -33,6 +43,12 @@ import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
 Import-Module "Microsoft.TeamFoundation.DistributedTask.Task.Deployment.Internal"
 Import-Module "Microsoft.TeamFoundation.DistributedTask.Task.Deployment.RemoteDeployment"
 
+# If testCertificate option is not set, set it to default 
+if([string]::IsNullOrEmpty($testCertificate))
+{
+	$testCertificate = $defaultTestCertificateValue
+}
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1]: Remove this it is not required

File: Tasks/SqlServerDacpacDeployment/task.json [hunk 8]
```diff
@@ -14,12 +14,17 @@
     "version": {
         "Major": 1,
         "Minor": 0,
-        "Patch": 3
+        "Patch": 4
     },
     "demands": [
     ],
     "minimumAgentVersion": "1.88.0",
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/task.json]: Task min agent version to be 90

File: Tasks/IISWebAppDeployment/task.json (hunk 3 above)
Comment (chshrikh) [inline Tasks/IISWebAppDeployment/task.json]: I have send it same PR with code flow please have a look

File: Tasks/SqlServerDacpacDeployment/task.json [hunk 9]
```diff
@@ -32,39 +37,78 @@
...
+			    "defaultValue": "",
+			    "required": false,
+			    "helpMarkDown": "Provide administrator user password for connecting to target machine."
+		    },
+		    {
+          "name": "Protocol",
+          "type": "radio",
+          "label": "Protocol",
+          "required": false,
+          "defaultValue": "",
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/task.json]: indentation

File: Tasks/SqlServerDacpacDeployment/task.json [hunk 10]
```diff
@@ -32,39 +37,78 @@
...
+          "type": "radio",
+          "label": "Select Machines By",
+          "required": false,
+          "defaultValue": "machineNames",
+          "visibleRule": "Protocol = Invalid",
+          "options": {
+            "machineNames": "Machine Names",
+            "tags": "Tags"
+          },
+          "helpMarkDown": "Optionally, select a subset of machines either by providing machine names or tags."
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/task.json]: fix the indentation

File: Tasks/SqlServerDacpacDeployment/task.json [hunk 11]
```diff
@@ -32,39 +37,78 @@
...
-            "type": "string",
-            "label": "Deploy to Machines",
-            "required": false,
-            "defaultValue": "",
-            "helpMarkDown": "Provide a list of machines like, dbserver.fabrikam.com, webserver.fabrikam.com, 192.168.12.34, or tags like, Role:DB; OS:Win8.1. If multiple tags are provided, then the task will run in all the machines with the specified tags. For Azure Resource Groups, provide the virtual machine's name like, ffweb, ffdb. The default is to run the task in all machines of the machine/resource group."
+          "name": "MachineFilter",
+          "type": "string",
+          "label": "Deploy to Machines",
+          "required": false,
+          "defaultValue": "",
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/task.json]: fix wherever applicable

File: Tasks/SqlServerDacpacDeployment/task.json [hunk 12]
```diff
@@ -32,39 +37,78 @@
...
+          "visibleRule": "Protocol = Invalid",
+          "helpMarkDown": "Provide a list of machines like, dbserver.fabrikam.com, webserver.fabrikam.com, 192.168.12.34, or tags like, Role:DB; OS:Win8.1. If multiple tags are provided, then the task will run in all the machines with the specified tags. For Azure Resource Groups, provide the virtual machine's name like, ffweb, ffdb. The default is to run the task in all machines of the machine/resource group."
         },
         {
             "name": "DacpacFile",
             "type": "string",
             "label": "DACPAC File",
             "required": true,
+		      	"groupName": "deployment",
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/task.json]: here as well

File: Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1 [hunk 13]
```diff
@@ -1,5 +1,9 @@
 ﻿param (
     [string]$environmentName,
+	[string]$adminUserName,
+    [string]$adminPassword,
+    [string]$protocol,
+    [string]$testCertificate,
```
Comment (mvvsubbu) [inline Tasks/SqlServerDacpacDeployment/DeployToSqlServer.ps1]: change this to testCert

**Pull request merged**